WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -49,6 +64,12 @@ public TorMonitor(TimeSpan period, Uri fallbackBackendUri, TorProcessManager tor
 
 	private Task? BootstrapTask { get; set; }
 
+	/// <summary>Whether we should try Tor process restart to fix <see cref="ReplyType.TtlExpired"/> issues.</summary>
+	private bool TryTorRestart { get; set; } = true;
+
+	/// <remarks>Assignment and cancel operations must be guarded with <see cref="_lock"/>.</remarks>
+	private CancellationTokenSource? ForceTorRestartCts { get; set; }
```

Comment:
I'm not sure this is correct.

It's been assigned in `MonitorEventsAsync`, to a variable that's disposed at the end of the function. So when we want to cancel it later, then it's gonna be disposed?

Reply:
I guess what I was missing is that `await foreach (TorControlReply reply in torControlClient.ReadEventsAsync(linkedCts2.Token).ConfigureAwait(false))` is waiting for a long time until an event comes right?